packages/xdl/src/Exp.js

```diff
@@ -44,87 +45,64 @@ export async function determineEntryPointAsync(root: string) {
-  let name = opts.name;
-  let root = path.join(selectedDir, name);
-
-  Analytics.logEvent('New Project', {
-    selectedDir,
-    name,
-  });
-
-  let stats;
-  try {
-    // If file doesn't exist it will throw an error.
-    // Don't want to continue unless there is nothing there.
-    stats = fs.statSync(root);
-  } catch (e) {
-    stats = null;
-  }
-  // This check is required because without it, the retry button would throw an error because the directory already exists,
-  // even though it is empty.
-  if (stats && !(stats.isDirectory() && fs.readdirSync(root).length === 0)) {
-    throw new XDLError(
-      ErrorCode.DIRECTORY_ALREADY_EXISTS,
-      `The path "${root}" already exists.\nPlease choose a different parent directory or project name.`
-    );
-  }
-
-  // Download files
-  let starterAppPath = await _downloadStarterAppAsync(
-    templateId,
-    opts.progressFunction,
-    opts.retryFunction
-  );
-  return { starterAppPath, name, root };
-}
-
-export async function extractTemplateApp(starterAppPath: string, name: string, root: string) {
+export async function extractTemplateApp(
+  templateSpec: string | object,
+  name: string,
+  projectRoot: string,
+  packageManager: 'yarn' | 'npm' = 'npm'
+) {
   Logger.notifications.info({ code: NotificationCode.PROGRESS }, MessageCode.EXTRACTING);
-  await fs.mkdirp(root);
-  await Extract.extractAsync(starterAppPath, root);
+  await pacote.extract(templateSpec, projectRoot, {
+    cache: path.join(UserSettings.dotExpoHomeDirectory(), 'template-cache'),
+  });
 
   // Update files
   Logger.notifications.info({ code: NotificationCode.PROGRESS }, MessageCode.CUSTOMIZING);
 
-  // Update app.json
-  let appJson = await fs.readFile(path.join(root, 'app.json'), 'utf8');
-  let customAppJson = appJson
-    .replace(/\"My New Project\"/, `"${name}"`)
-    .replace(/\"my-new-project\"/, `"${name}"`);
-  await fs.writeFile(path.join(root, 'app.json'), customAppJson, 'utf8');
+  let appFile = new JsonFile(path.join(projectRoot, 'app.json'));
+  let appJson = await appFile.readAsync();
+  appJson.expo = { ...appJson.expo, name, slug: name };
+  await appFile.writeAsync(appJson);
+
+  let packageFile = new JsonFile(path.join(projectRoot, 'package.json'));
+  let packageJson = await packageFile.readAsync();
+  // Adding `private` stops npm from complaining about missing `name` and `version` fields.
+  // We don't add a `name` field because it also exists in `app.json`.
+  packageJson.private = true;
+  // These are metadata fields related to the template package, let's remove them from the package.json.
+  delete packageJson.name;
+  delete packageJson.version;
+  delete packageJson.description;
+  delete packageJson.tags;
+  delete packageJson.repository;
+  // pacote adds these, but we don't want them in the package.json of the project.
+  delete packageJson._resolved;
+  delete packageJson._integrity;
+  delete packageJson._from;
```

Comment:
You're right to be careful about mutation, since non-local mutation can cause trouble (e.g. if I was mutating an object passed to this function as an argument), but in this case it's local mutation, which is completely fine.

I don't use `lodash` for things that can be done easily with plain JS. `_.omit` will also be harder to correctly type with TypeScript or Flow.

However, since you pointed this out, I've now changed the code to defensively clone the object before mutating it to make the intent (mutating only locally) clearer.

Reply:
My thinking with regards to types was that using `delete` would help code like this produce a type error:
```js
let data = { a: 1, b: 2 };
delete data.a;
console.log(data.a + data.b);
```
However, it turns out this code passes the type check in both TypeScript and Flow (bummer!) so my argument is moot. 😊 

I think we're in agreement about `lodash`. But, in this particular case fewer lines of code doesn't bring much value (I actually prefer having these properties on separate lines, this gives better diffs in case fields get added/removed) and I think `delete` here is also quite explicit.